pithos/pithos.py

```diff
@@ -862,30 +915,29 @@ def on_gst_volume(self, player, volumespec):
     def on_gst_source(self, player, params):
         """ Setup httpsoupsrc to match Pithos proxy settings """
         soup = player.props.source.props
-        proxy = self.get_proxy()
-        if proxy and hasattr(soup, 'proxy'):
-            scheme, user, password, hostport = parse_proxy(proxy)
-            soup.proxy = hostport
-            soup.proxy_id = user
-            soup.proxy_pw = password
+        if hasattr(soup, 'proxy'):
+            soup.timeout = 600
+            proxy = self.get_proxy()
+            if proxy:
+                scheme, user, password, hostport = parse_proxy(proxy)
+                soup.proxy = hostport
+                soup.proxy_id = user
+                soup.proxy_pw = password
 
     def song_text(self, song):
         title = html.escape(song.title)
         artist = html.escape(song.artist)
         album = html.escape(song.album)
         msg = []
         if song is self.current_song:
-            song.position = self.query_position()
             if not song.bitrate is None:
                 msg.append("%skbit/s" % (song.bitrate))
-
-            if song.position is not None and song.duration is not None:
-                pos_str = self.format_time(song.position)
-                msg.append("%s / %s" % (pos_str, song.duration_message))
-                if self.playing == False:
-                    msg.append("Paused")
-            if self.player_status.buffer_percent < 100:
-                msg.append("Buffering (%i%%)" % self.player_status.buffer_percent)
+            pos_str = self.format_time(self.query_position() or 0)
+            msg.append("%s / %s" % (pos_str, song.duration_message))
+            if self.playing == False:
```

Comment:
__[flake8]__
* [E712] comparison to False should be 'if cond is False:' or 'if not cond:'

<!-- Comment by [SideCI](https://sideci.com) -->